waku/v2/protocol/waku_rln_relay/rln.nim

```diff
@@ -17,72 +17,100 @@ elif defined(MacOsX):
 # all the following procedures are Nim wrappers for the functions defined in libName
 {.push dynlib: libName, raises: [Defect].}
 
-
 ## Buffer struct is taken from
 # https://github.com/celo-org/celo-threshold-bls-rs/blob/master/crates/threshold-bls-ffi/src/ffi.rs
 type Buffer* = object
   `ptr`*: ptr uint8
   len*: uint
-  
+
 #------------------------------ Merkle Tree operations -----------------------------------------
-proc update_next_member*(ctx: RLN[Bn256],
-                        input_buffer: ptr Buffer): bool {.importc: "update_next_member".}
+proc update_next_member*(ctx: ptr RLN, input_buffer: ptr Buffer): bool {.importc: "set_next_leaf".}
+## adds an element in the merkle tree to the next available position
 ## input_buffer points to the id commitment byte seq
 ## the return bool value indicates the success or failure of the operation
 
-proc delete_member*(ctx: RLN[Bn256], index: uint): bool {.importc: "delete_member".}
+proc delete_member*(ctx: ptr RLN, index: uint): bool {.importc: "delete_leaf".}
 ## index is the position of the id commitment key to be deleted from the tree
 ## the deleted id commitment key is replaced with a zero leaf
 ## the return bool value indicates the success or failure of the operation
 
-proc get_root*(ctx: RLN[Bn256], output_buffer: ptr Buffer): bool {.importc: "get_root".}
+proc get_root*(ctx: ptr RLN, output_buffer: ptr Buffer): bool {.importc: "get_root".}
 ## get_root populates the passed pointer output_buffer with the current tree root
 ## the output_buffer holds the Merkle tree root of size 32 bytes
 ## the return bool value indicates the success or failure of the operation
-## 
+
+proc get_merkle_proof*(ctx: ptr RLN, index: uint, output_buffer: ptr Buffer): bool {.importc: "get_proof".}
```

Comment:
I'd probably keep this as qualified merkle proof, as "get proof" could be a ZK proof

Reply:
So is fine like this, or you mean changing the zerokit RLN api name to get_merkle_proof too?

Reply:
`get_merkle_proof`

Reply:
This nim wrapper is already named `get_merkle_proof` and with this name the API is used internally in nwaku. If you mean to change zerokit RLN public API name, this has to be addressed with an issue in vacp2p/zerokit.